Sodium/Sodium.swift

```diff
@@ -17,15 +21,16 @@ public struct Sodium {
     public let secretStream = SecretStream()
     public let aead = Aead()
 
-    public init() {
-        _ = Sodium.once
+    public init() throws {
+        let code = Sodium.initCode
+        guard code >= 0 else {
+            throw InitError.code(code)
+        }
     }
 }
 
 extension Sodium {
-    private static let once: Void = {
-        guard sodium_init() >= 0 else {
-            fatalError("Failed to initialize libsodium")
-        }
+    private static let initCode: Int32 = {
+        return sodium_init()
```

Comment:
`sodium_init()` returns a `Cint`, not an `Int32`. But the return code can only be `-1` on error.

Reply:
Replaced with `CInt`. It is `Int32` on my mac (unsure about Linux, but probably that's independent)

Reply:
I kept the previous check for `code >= 0`. Should we change it to `if code == -1 { throw InitError.code(-1) }` ?

Reply:
No, but what I meant is that the return code is not something worth propagating in `InitError`. The code is always going to be `-1`.